### PR TITLE
Use redux in react components

### PR DESCRIPTION
**Pull request by moise-mulungu**
At this step, I:
- Kept the state immutable
- Imported redux Provider
- Used useSelector, useDispatch
- Added functionalities to the app's components
- Updated the README file

### CODE REVIEW

File: src/components/Book.js [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
 import Proptypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { del } from '../redux/books/books';
 
 export default function Book({ book }) {
+  const dispatch = useDispatch();
   return (
     <div className="bookCard">
       <div className="">
+        <p className="genre">{book.genre}</p>
```
Comment (DeliceLydia) [inline src/components/Book.js:11]: - [x] The page is not rendered check the issue below  the issue  The prop `books` is marked as required in `Books`, but its value is `undefined`
dgf

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @moise-mulungu, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! Highlights for your project:
- No linter errors
- Followed git workflow


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: src/index.js [hunk 2]
```diff
@@ -1,11 +1,15 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import { BrowserRouter } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import configureStore from './redux/configureStore';
 import App from './App';
 
 ReactDOM.render(
```
Comment (rloterh) [inline src/index.js:8]: - [ ] OPTIONAL
Your app is not rendering as intended.

**Render error:**
image


To improve upon the above kindly update your _index.js_ file to look with React 18 syntax below:

```ruby
import { StrictMode } from 'react';
import { Provider } from 'react-redux';
import store from './redux/configureStore';
import App from './components/App';
import { createRoot } from "react-dom/client";

const rootElement = document.getElementById("root");
const root = createRoot(rootElement);

root.render(
  <Provider store={store}>
    <StrictMode>
      <App />
    </StrictMode>
  </Provider>,
);

```

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Moise, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾 :shipit:. Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have no more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**